File: src/main/java/de/maggiwuerze/xdccloader/model/entity/Channel.java
package de.maggiwuerze.xdccloader.model.entity;

import java.time.LocalDateTime;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
public class Channel {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	Long id;

	@Column(nullable = false)
	String name;

	@Column(nullable = false)
	LocalDateTime date = LocalDateTime.now();

	public Channel(String name) {
		this.name = name;
	}
}

File: src/main/kotlin/de/maggiwuerze/xdccwebloader/model/entity/Channel.kt
package de.maggiwuerze.xdccwebloader.model.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import java.time.LocalDateTime
import java.util.*

@Entity
class Channel(
    @Id
    var id: UUID = UUID.randomUUID(),

    @Column(nullable = false)
    var name: String,

    @Column(nullable = false)
    var date: LocalDateTime = LocalDateTime.now()
)